src/main/java/com/group8/library_management/mapper/BorrowRecordMapper.java

```diff
@@ -0,0 +1,24 @@
+package com.group8.library_management.mapper;
+
+import com.group8.library_management.dto.response.BorrowRecordResponse;
+import com.group8.library_management.entity.BorrowRecord;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BorrowRecordMapper {
+    public BorrowRecordResponse toResponse(BorrowRecord entity) {
+        String bookTitle = null;
+        if (entity.getBorrowRequestDetail() != null &&
+            entity.getBorrowRequestDetail().getCopy() != null &&
+            entity.getBorrowRequestDetail().getCopy().getBook() != null) {
+            bookTitle = entity.getBorrowRequestDetail().getCopy().getBook().getTitle();
+        }
```

Comment:
```suggestion
        String bookTitle = Optional.ofNullable(entity.getBorrowRequestDetail())
        .map(detail -> detail.getCopy())
        .map(copy -> copy.getBook())
        .map(book -> book.getTitle())
        .orElse(null);
```

em nghĩ sao? mà return null hay blank `""` sẽ hợp lý hơn

Reply:
Dạ em chọn phương án null ạ. Em nghĩ trong thực tế, đã mượn sách mà chưa có tên sách là không hợp lý. Nếu để null thì sẽ dễ log lỗi hơn, tách biệt rõ ràng giữa chưa nhập và đã nhập rỗng ạ